src/calendar/updater.js

```diff
@@ -46,6 +46,14 @@ export default function shouldComponentUpdate(nextProps, nextState) {
       field: 'current'
     };
   }
+
+  if (nextState.monthFormat !== this.state.monthFormat) {
```

Comment:
I don't see why this is necessary. The CalendarHeader is updating fine with the change in header/index.js